airmozilla/popcorn/static/popcorn/js/editor.js

```diff
@@ -1,2 +1,29 @@
-// Initialize the editor with the div id and path to Popcorn Editor.
-PopcornEditor.init('editor', '/static/popcorn/PopcornEditor/editor.html');
+$(document).ready(function () {
+  function getSlug() {
+    var config = $('slug');
+    if (config.length) {
+      return config.data('url');
+    } else {
+      throw 'Slug not provided in DOM. Cannot get slug';
+    }
+  }
+
+  PopcornEditor.listen(PopcornEditor.events.loaded, function () {
+    $.get('/popcorn/data/', {
+        slug: getSlug()
+    })
+    .done(function (response) {
+      if(response.data) {
+        PopcornEditor.loadInfo(response.data);
+      } else {
+        PopcornEditor.loadInfo(PopcornEditor.createTemplate(response.metadata));
+      }
+    })
+    .fail(function() {
+      console.warn("Unable to load popcorn data :(");
+      console.error.apply(console, arguments);
```

Comment:
This is on my TODO list but seeing as exporting edits also isn't yet implemented either I kind of put that on the back burner.